File: diary_sample/src/main/java/com/example/diary/feature/article/domain/Article.java
package com.example.diary.feature.article.domain;

import com.example.diary.feature.article.dto.ArticleCreateRequest;
import com.example.diary.feature.article.dto.ArticleServiceCreateRequest;
import com.example.diary.feature.image.domain.Image;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;


@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Article {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long articleId;

    private int views;
    private String title;
    private String content;
    @OneToMany(mappedBy = "article")
    private List<Image> images;

    @Builder
    private Article(int views, String title, String content, List<Image> images) {
        this.views = views;
        this.title = title;
        this.content = content;
        this.images = images;
    }

    public static Article create(ArticleServiceCreateRequest request) {
        return Article.builder()
                .title(request.getTitle())
                .content(request.getContent())
                .images(request.getImage())
                .views(0)
                .build();
    }

}

File: diary/src/main/kotlin/sample/diary/domain/article/Article.kt
package sample.diary.domain.article

import jakarta.persistence.*
import sample.diary.domain.image.Image
import sample.diary.dto.article.ArticleServiceRequest

@Entity
class Article (
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val articleId: Long,
    val title: String,
    val content: String,
    val view: Int,

    @OneToMany(mappedBy = "article", fetch = FetchType.LAZY)
    var imageList: MutableList<Image>
){
    constructor() : this(0,"", "", 0, mutableListOf())

    companion object {
        fun create(request: ArticleServiceRequest): Article {
            return Article(
                articleId = 0,
                title = request.title,
                content = request.content,
                view = 0,
                imageList = mutableListOf()
            )
        }
    }

}